tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,155 @@
+
+    def get_dict_element(self, data, path, delimiter='.'):
+        return functools.reduce(operator.getitem, path.split(delimiter), data)
+
+
+def main():
+    """
+        python rule_extractor --ip 127.0.0.1 --port 9090 --type rules
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-i',
+        '--ip',
+        dest='ip',
+        help='Specify the target control-plane ingress IP address'
+    )
+    parser.add_argument(
+        '-p',
+        '--port',
+        dest='port',
+        default='8443',
+        help='Specify the target control-plane ingress port'
+    )
+    parser.add_argument(
+        '-t',
+        '--type',
+        required=True,
+        dest='rule_type',
+        help='Specify the rule type. For example: rules or alerts'
+    )
+
+    args = parser.parse_args()
+    if not args.ip:
+        # Obtain VM ip by connecting to it and running a command
+        # Requires that the VM is up and running.
+        ip = subprocess.check_output(
+            [
+                "vagrant",
+                "ssh",
+                "--command",
+                "hostname -I | cut -d' ' -f1"
+            ]
+        ).decode(sys.stdout.encoding).strip()
+        # Validate that we got a real ipv4 address
+        try:
+            ipaddress.ip_address(ip)
+        except ValueError:
+            sys.stderr.write("Ip address: {} is invalid".format(ip))
+        except Exception as exc:
+            sys.stderr.write(
+                "Syntax error while checking ip address validity: {!s}".format(
+                    exc
+                )
+            )
+        alertrule = AlertRuleFile(ip=ip, port=args.port)
+    else:
+        alertrule = AlertRuleFile(ip=args.ip, port=args.port)
+
+    if args.rule_type == "rules":
+        # Returns list of rules that are currently loaded
+        out = alertrule.query_prometheus_api(route="rules")
+        alertrule.write_to_file(out, args.rule_type)
+        # For documentation purposes, we don't want to litter the whole thing
+        # with all fields, so lets strip off some "ambiguous" fields
+        rule_group = alertrule.get_dict_element(out, "data.groups")
+        recording_rules = {}
+        alerting_rules = {}
+        for item in rule_group:
+            if 'rules' in item:
+                for rule in item['rules']:
+                    # let's format things nicely
+                    # rule type can be alerting or recording
+                    if rule['type'] == "recording":
+                        fixup_record_rule = {}
+                        fixup_record_rule['name'] = rule['name']
+                        fixup_record_rule['query'] = rule['query']
+                        rule_name = rule['name']
+                        if rule_name not in recording_rules:
+                            recording_rules[rule_name] = {}
+                        recording_rules[rule_name].update(fixup_record_rule)
```

Comment:
Could be replaced by:
```suggestion
                        recording_rules.setdefault(rule['name'], {}).update({
                            'name': rule['name'],
                            'query': rule['query'],
                        })
```
But it should even be:
```
                        recording_rules[rule['name']] = {
                            'name': rule['name'],
                            'query': rule['query'],
                        }
```